# Importation de la bibbliotheque panda et numpy

In [2]:
import pandas as pd
import numpy as np


# Importation de notre dataset

In [8]:
df = pd.read_csv("student-mat.csv",sep=";")

# 2- Description du dataset

In [11]:
df 

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[395 rows x 33 columns]

# Origine du dataset: Le dataset appelé student-mat , est un dataset qui prend son origine sur la collecte des donnees a caractere personnelle des etudiants de deux etablissements .

# Circonstances de creation de ce dataset: Les donnes ont ete recueillis chez les etudiants a partir soit d'un googe form,soit d'une descente faite sur le terrain.
.

# Previson : - A partir de ce data on peut faire des previsions sur la forme d'education des eleves ayant une moyenne assez basse en ce basant sur les conditions de vie de leurs parents etc....
# - On peut faire des previsions sur l'age moyen des filles de ces etablissements .

# Aperçu du dataset: On peut remarquer dans notre data a premiere vu que y'a pas de valeurs manquantes, mais on a une presence enorme de valeurs qualitatives. Et comme dimension matricielle on a 395 x 33.

# Ce qui concerne la qualite de notre dataset, nous pouvons dire que les donnes ont ete bien recoltes, grace a la minimisation des valeurs manquantes et on a une bonne structuration des donnees.

# 3- Affichage des 10 premieres ligne du dataset

In [12]:
df.head(10)

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4   at_home   teacher   
1     GP   F   17       U     GT3       T     1     1   at_home     other   
2     GP   F   15       U     LE3       T     1     1   at_home     other   
3     GP   F   15       U     GT3       T     4     2    health  services   
4     GP   F   16       U     GT3       T     3     3     other     other   
5     GP   M   16       U     LE3       T     4     3  services     other   
6     GP   M   16       U     LE3       T     2     2     other     other   
7     GP   F   17       U     GT3       A     4     4     other   teacher   
8     GP   M   15       U     LE3       A     3     2  services     other   
9     GP   M   15       U     GT3       T     3     4     other     other   

   ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0  ...      4        3      4     1     1      3        6   5   6   6  
1  ...      5        3      3     1     1      3        4   5   5   6  
2  ...      4        3      2     2     3      3       10   7   8  10  
3  ...      3        2      2     1     1      5        2  15  14  15  
4  ...      4        3      2     1     2      5        4   6  10  10  
5  ...      5        4      2     1     2      5       10  15  15  15  
6  ...      4        4      4     1     1      3        0  12  12  11  
7  ...      4        1      4     1     1      1        6   6   5   6  
8  ...      4        2      2     1     1      1        0  16  18  19  
9  ...      5        5      1     1     1      5        0  14  15  15  

[10 rows x 33 columns]

# 4-Affichage des 10 dernieres lignes du dataset

In [13]:
df.tail(10)

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
385     MS   F   18       R     GT3       T     2     2   at_home     other   
386     MS   F   18       R     GT3       T     4     4   teacher   at_home   
387     MS   F   19       R     GT3       T     2     3  services     other   
388     MS   F   18       U     LE3       T     3     1   teacher  services   
389     MS   F   18       U     GT3       T     1     1     other     other   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
385  ...      5        3      3     1     3      4        2  10   9  10  
386  ...      4        4      3     2     2      5        7   6   5   6  
387  ...      5        4      2     1     2      5        0   7   5   0  
388  ...      4        3      4     1     1      1        0   7   9   8  
389  ...      1        1      1     1     1      5        0   6   5   0  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[10 rows x 33 columns]

# 4-Affichage des colonnes de notre dataset

In [14]:
df.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

# 5-Mise en forme des titres des colonnes en majiscules

In [15]:
df.columns.str.upper()

Index(['SCHOOL', 'SEX', 'AGE', 'ADDRESS', 'FAMSIZE', 'PSTATUS', 'MEDU', 'FEDU',
       'MJOB', 'FJOB', 'REASON', 'GUARDIAN', 'TRAVELTIME', 'STUDYTIME',
       'FAILURES', 'SCHOOLSUP', 'FAMSUP', 'PAID', 'ACTIVITIES', 'NURSERY',
       'HIGHER', 'INTERNET', 'ROMANTIC', 'FAMREL', 'FREETIME', 'GOOUT', 'DALC',
       'WALC', 'HEALTH', 'ABSENCES', 'G1', 'G2', 'G3'],
      dtype='object')

# 7-affichons la taille de notre dataset

In [16]:
df.shape

(395, 33)

# 8-Affichons les informations sur les types de donnees de notre dataset

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 19  nursery     395 non-null    object
 20  higher    

# Affichage des types de donnees

In [92]:
df.dtypes

school        object
sex           object
age            int64
address       object
famsize       object
Pstatus       object
Medu           int64
Fedu           int64
Mjob          object
Fjob          object
reason        object
guardian      object
traveltime     int64
studytime      int64
failures       int64
schoolsup     object
famsup        object
paid          object
activities    object
nursery       object
higher        object
internet      object
romantic      object
famrel         int64
freetime       int64
goout          int64
Dalc           int64
Walc           int64
health         int64
absences       int64
G1             int64
G2             int64
G3             int64
dtype: object

# Verifications de valeurs manquantes

In [19]:
df.isnull().sum()

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64

# Ce dataset ne presente aucune valeur manquante

# 9-Affichons les statistiques sur les colonnes

In [20]:
df.describe()

age        Medu        Fedu  traveltime   studytime    failures  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean    16.696203    2.749367    2.521519    1.448101    2.035443    0.334177   
std      1.276043    1.094735    1.088201    0.697505    0.839240    0.743651   
min     15.000000    0.000000    0.000000    1.000000    1.000000    0.000000   
25%     16.000000    2.000000    2.000000    1.000000    1.000000    0.000000   
50%     17.000000    3.000000    2.000000    1.000000    2.000000    0.000000   
75%     18.000000    4.000000    3.000000    2.000000    2.000000    0.000000   
max     22.000000    4.000000    4.000000    4.000000    4.000000    3.000000   

           famrel    freetime       goout        Dalc        Walc      health  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean     3.944304    3.235443    3.108861    1.481013    2.291139    3.554430   
std      0.896659    0.998862    1.113278    0.890741    1.287897    1.390303   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    3.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

         absences          G1          G2          G3  
count  395.000000  395.000000  395.000000  395.000000  
mean     5.708861   10.908861   10.713924   10.415190  
std      8.003096    3.319195    3.761505    4.581443  
min      0.000000    3.000000    0.000000    0.000000  
25%      0.000000    8.000000    9.000000    8.000000  
50%      4.000000   11.000000   11.000000   11.000000  
75%      8.000000   13.000000   13.000000   14.000000  
max     75.000000   19.000000   19.000000   20.000000

# Interpretation la statistique 

# L'attribut describe se base sur les donnees a valeurs quantitatives.Comme nombre de colonnes indexees on a 16 colonnes , pour un type de donnees qui est int64. 

# Correlation obtenue: les données dans notre data sont fortement concentrées de chaque colonne.


# Interpretation de la valeur maximale par rapport a la distribution moyenne: les donnees recoltees tendent vers la moyenne.

# 10-Transposons le dataset

In [24]:
df_transpose = df.transpose()

In [25]:
df_transpose

0        1        2         3       4           5       6    \
school           GP       GP       GP        GP      GP          GP      GP   
sex               F        F        F         F       F           M       M   
age              18       17       15        15      16          16      16   
address           U        U        U         U       U           U       U   
famsize         GT3      GT3      LE3       GT3     GT3         LE3     LE3   
Pstatus           A        T        T         T       T           T       T   
Medu              4        1        1         4       3           4       2   
Fedu              4        1        1         2       3           3       2   
Mjob        at_home  at_home  at_home    health   other    services   other   
Fjob        teacher    other    other  services   other       other   other   
reason       course   course    other      home    home  reputation    home   
guardian     mother   father   mother    mother  father      mother  mother   
traveltime        2        1        1         1       1           1       1   
studytime         2        2        2         3       2           2       2   
failures          0        0        3         0       0           0       0   
schoolsup       yes       no      yes        no      no          no      no   
famsup           no      yes       no       yes     yes         yes      no   
paid             no       no      yes       yes     yes         yes      no   
activities       no       no       no       yes      no         yes      no   
nursery         yes       no      yes       yes     yes         yes     yes   
higher          yes      yes      yes       yes     yes         yes     yes   
internet         no      yes      yes       yes      no         yes     yes   
romantic         no       no       no       yes      no          no      no   
famrel            4        5        4         3       4           5       4   
freetime          3        3        3         2       3           4       4   
goout             4        3        2         2       2           2       4   
Dalc              1        1        2         1       1           1       1   
Walc              1        1        3         1       2           2       1   
health            3        3        3         5       5           5       3   
absences          6        4       10         2       4          10       0   
G1                5        5        7        15       6          15      12   
G2                6        5        8        14      10          15      12   
G3                6        6       10        15      10          15      11   

                7         8       9    ...      385         386       387  \
school           GP        GP      GP  ...       MS          MS        MS   
sex               F         M       M  ...        F           F         F   
age              17        15      15  ...       18          18        19   
address           U         U       U  ...        R           R         R   
famsize         GT3       LE3     GT3  ...      GT3         GT3       GT3   
Pstatus           A         A       T  ...        T           T         T   
Medu              4         3       3  ...        2           4         2   
Fedu              4         2       4  ...        2           4         3   
Mjob          other  services   other  ...  at_home     teacher  services   
Fjob        teacher     other   other  ...    other     at_home     other   
reason         home      home    home  ...    other  reputation    course   
guardian     mother    mother  mother  ...   mother      mother    mother   
traveltime        2         1       1  ...        2           3         1   
studytime         2         2       2  ...        3           1         3   
failures          0         0       0  ...        0           0         1   
schoolsup       yes        no      no  ...       no          no        no   
famsup          yes 

In [93]:
df.T

0        1        2         3       4           5       6    \
school           GP       GP       GP        GP      GP          GP      GP   
sex               F        F        F         F       F           M       M   
age              18       17       15        15      16          16      16   
address           U        U        U         U       U           U       U   
famsize         GT3      GT3      LE3       GT3     GT3         LE3     LE3   
Pstatus           A        T        T         T       T           T       T   
Medu              4        1        1         4       3           4       2   
Fedu              4        1        1         2       3           3       2   
Mjob        at_home  at_home  at_home    health   other    services   other   
Fjob        teacher    other    other  services   other       other   other   
reason       course   course    other      home    home  reputation    home   
guardian     mother   father   mother    mother  father      mother  mother   
traveltime        2        1        1         1       1           1       1   
studytime         2        2        2         3       2           2       2   
failures          0        0        3         0       0           0       0   
schoolsup       yes       no      yes        no      no          no      no   
famsup           no      yes       no       yes     yes         yes      no   
paid             no       no      yes       yes     yes         yes      no   
activities       no       no       no       yes      no         yes      no   
nursery         yes       no      yes       yes     yes         yes     yes   
higher          yes      yes      yes       yes     yes         yes     yes   
internet         no      yes      yes       yes      no         yes     yes   
romantic         no       no       no       yes      no          no      no   
famrel            4        5        4         3       4           5       4   
freetime          3        3        3         2       3           4       4   
goout             4        3        2         2       2           2       4   
Dalc              1        1        2         1       1           1       1   
Walc              1        1        3         1       2           2       1   
health            3        3        3         5       5           5       3   
absences          6        4       10         2       4          10       0   
G1                5        5        7        15       6          15      12   
G2                6        5        8        14      10          15      12   
G3                6        6       10        15      10          15      11   

                7         8       9    ...      385         386       387  \
school           GP        GP      GP  ...       MS          MS        MS   
sex               F         M       M  ...        F           F         F   
age              17        15      15  ...       18          18        19   
address           U         U       U  ...        R           R         R   
famsize         GT3       LE3     GT3  ...      GT3         GT3       GT3   
Pstatus           A         A       T  ...        T           T         T   
Medu              4         3       3  ...        2           4         2   
Fedu              4         2       4  ...        2           4         3   
Mjob          other  services   other  ...  at_home     teacher  services   
Fjob        teacher     other   other  ...    other     at_home     other   
reason         home      home    home  ...    other  reputation    course   
guardian     mother    mother  mother  ...   mother      mother    mother   
traveltime        2         1       1  ...        2           3         1   
studytime         2         2       2  ...        3           1         3   
failures          0         0       0  ...        0           0         1   
schoolsup       yes        no      no  ...       no          no        no   
famsup          yes 

# 11- Affichage de la colonne age 

In [27]:
age =df["age"]

age

0      18
1      17
2      15
3      15
4      16
       ..
390    20
391    17
392    21
393    18
394    19
Name: age, Length: 395, dtype: int64

# 12-Affichage des informations suivantes concernant la colonne age: moyenne,minimum et ecart type

In [29]:
age.mean()


16.696202531645568

In [30]:
age.max()

22

In [31]:
age.min()

15

In [32]:
age.std()

1.2760427246056245

# 13- aAffichage de toutes ces statiques avec une seule commande

In [33]:
age.describe()

count    395.000000
mean      16.696203
std        1.276043
min       15.000000
25%       16.000000
50%       17.000000
75%       18.000000
max       22.000000
Name: age, dtype: float64

# 14-Extraction d'une mini base de donnees constituee des colonnes "sex","age","studytime","G1","G2","G3"

In [86]:
sex =df["sex"]
age = df["age"]
studytime = df["studytime"]
G1 = df["G1"]
G2 = df["G2"]
G3 = df["G3"]

mini_base = pd.concat([sex,age,studytime,G1,G2,G3], axis=1)

mini_base



sex  age  studytime  G1  G2  G3
0     F   18          2   5   6   6
1     F   17          2   5   5   6
2     F   15          2   7   8  10
3     F   15          3  15  14  15
4     F   16          2   6  10  10
..   ..  ...        ...  ..  ..  ..
390   M   20          2   9   9   9
391   M   17          1  14  16  16
392   M   21          1  10   8   7
393   M   18          1  11  12  10
394   M   19          1   8   9   9

[395 rows x 6 columns]

# 15-Affichons les statistiques sur cette mini base

In [36]:
mini_base.describe()

age   studytime          G1          G2          G3
count  395.000000  395.000000  395.000000  395.000000  395.000000
mean    16.696203    2.035443   10.908861   10.713924   10.415190
std      1.276043    0.839240    3.319195    3.761505    4.581443
min     15.000000    1.000000    3.000000    0.000000    0.000000
25%     16.000000    1.000000    8.000000    9.000000    8.000000
50%     17.000000    2.000000   11.000000   11.000000   11.000000
75%     18.000000    2.000000   13.000000   13.000000   14.000000
max     22.000000    4.000000   19.000000   19.000000   20.000000

# La colonne sex n'apparait pas , car elle contient des valeurs des valeurs qualitatives qui sont M(masculin) et F(Feminin)

# 16-Affichons le nombres de garcons et de filles de la mini-base

In [41]:
mini_base["sex"].value_counts()

F    208
M    187
Name: sex, dtype: int64

# Nous avons 208 filles et 187 garcons

# 17-Affichons les informatiosn sur l'etudiant a l'indice4

In [95]:
 df[4:5]



school sex  age address famsize Pstatus  Medu  Fedu   Mjob   Fjob  ...  \
4     GP   F   16       U     GT3       T     3     3  other  other  ...   

  famrel freetime  goout  Dalc  Walc health absences G1  G2  G3  
4      4        3      2     1     2      5        4  6  10  10  

[1 rows x 33 columns]

# 18-Selectionnons dans la mini base de donnee les etudiants allant de l'indice 10 a l'indice 19

In [97]:
etudiant = df[10:20]
etudiant

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
10     GP   F   15       U     GT3       T     4     4   teacher    health   
11     GP   F   15       U     GT3       T     2     1  services     other   
12     GP   M   15       U     LE3       T     4     4    health  services   
13     GP   M   15       U     GT3       T     4     3   teacher     other   
14     GP   M   15       U     GT3       A     2     2     other     other   
15     GP   F   16       U     GT3       T     4     4    health     other   
16     GP   F   16       U     GT3       T     4     4  services  services   
17     GP   F   16       U     GT3       T     3     3     other     other   
18     GP   M   17       U     GT3       T     3     2  services  services   
19     GP   M   16       U     LE3       T     4     3    health     other   

    ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
10  ...      3        3      3     1     2      2        0  10   8   9  
11  ...      5        2      2     1     1      4        4  10  12  12  
12  ...      4        3      3     1     3      5        2  14  14  14  
13  ...      5        4      3     1     2      3        2  10  10  11  
14  ...      4        5      2     1     1      3        0  14  16  16  
15  ...      4        4      4     1     2      2        4  14  14  14  
16  ...      3        2      3     1     2      2        6  13  14  14  
17  ...      5        3      2     1     1      4        4   8  10  10  
18  ...      5        5      5     2     4      5       16   6   5   5  
19  ...      3        1      3     1     3      5        4   8  10  10  

[10 rows x 33 columns]

# 19-Affichons  la note G1 ,G2,G3 pour les etudiants precedemment selectionnes

In [98]:
pd.concat([etudiant["G1"],etudiant["G2"],etudiant["G3"]],axis=1)

G1  G2  G3
10  10   8   9
11  10  12  12
12  14  14  14
13  10  10  11
14  14  16  16
15  14  14  14
16  13  14  14
17   8  10  10
18   6   5   5
19   8  10  10

# 20-Affichons les statistiques sur la selection

In [57]:
etudiant.describe()

age  studytime         G1      G2         G3
count  10.000000  10.000000  10.000000  10.000  10.000000
mean   15.600000   1.900000  10.700000  11.300  11.500000
std     0.699206   0.875595   2.907844   3.335   3.205897
min    15.000000   1.000000   6.000000   5.000   5.000000
25%    15.000000   1.000000   8.500000  10.000  10.000000
50%    15.500000   2.000000  10.000000  11.000  11.500000
75%    16.000000   2.750000  13.750000  14.000  14.000000
max    17.000000   3.000000  14.000000  16.000  16.000000

# 21-Affichons les notes des etudiantes

In [99]:
etudiante = df[(df["sex"]=="F")]

G1 = etudiante["G1"]

G2 = etudiante["G2"]

G3 = etudiante["G3"]

pd.concat([G1,G2,G3],axis=1)

G1  G2  G3
0     5   6   6
1     5   5   6
2     7   8  10
3    15  14  15
4     6  10  10
..   ..  ..  ..
385  10   9  10
386   6   5   6
387   7   5   0
388   7   9   8
389   6   5   0

[208 rows x 3 columns]

# 22-Affichons les notes et l'age des etudiantes qui etudient 2 heures par jour

In [102]:
set = df[(df["sex"]=="F")&(df["studytime"]==2)]
pd.concat([set["G1"],set["G2"],set["G3"],set["age"]],axis=1)
    

G1  G2  G3  age
0     5   6   6   18
1     5   5   6   17
2     7   8  10   15
4     6  10  10   16
7     6   5   6   17
..   ..  ..  ..  ...
377   8   9  10   18
378  15  15  15   18
379  10  10  10   17
388   7   9   8   18
389   6   5   0   18

[113 rows x 4 columns]

# 23-Affichons les occurences des metiers des papas

In [101]:
df["Fjob"].value_counts()

other       217
services    111
teacher      29
at_home      20
health       18
Name: Fjob, dtype: int64

# 24-Affichons sur le dataset d'origine un filtre permettant de d'afficher les notes des etudiantes dont le pere est soit medecin soit des papas

In [125]:
data = df[(df["sex"]=="F") & ((df["Fjob"]=="health") | (df["Fjob"]=="teacher")) ]

pd.concat([data["G1"],data["G2"],data["G3"]],axis=1)


G1  G2  G3
0     5   6   6
7     6   5   6
10   10   8   9
24   10   9   8
38   12  12  11
49    7   7   7
60   10  11  11
63   10   9   9
105  10  11  11
109  14  15  16
122  13  13  13
130  12   0   0
139  16  16  15
154  11  11  12
169  14  14  14
198  18  18  18
199   9   9  10
222  16  16  17
274  10  10  10
278   9   8   8
287  13  12  12
303  17  17  18
314  15  13  13
329  14  14  14
369  14  12  11

# 25- Renommons dans la mini base les colonnes G1,G2,G3 respectivement en Maths ,Anglais,Francais

In [117]:
mini_base = mini_base.rename(columns={"G1":"Maths","G2":"Anglais","G3":"Français"})

mini_base

sex age  studytime  Maths  Anglais  Français
0     F  18          2      5        6         6
1     F  17          2      5        5         6
2     F  18          2      7        8        10
3     F  15          3     15       14        15
4     F  16          2      6       10        10
..   ..  ..        ...    ...      ...       ...
390   M  20          2      9        9         9
391   M  17          1     14       16        16
392   M  21          1     10        8         7
393   M  18          1     11       12        10
394   M  19          1      8        9         9

[395 rows x 6 columns]

# 26-Modifions l'age de l'etudiant a l'indice 2 en lui donnant 18 ans 

In [87]:
mini_base.loc[2,"age"] = "18"

mini_base

sex age  studytime  G1  G2  G3
0     F  18          2   5   6   6
1     F  17          2   5   5   6
2     F  18          2   7   8  10
3     F  15          3  15  14  15
4     F  16          2   6  10  10
..   ..  ..        ...  ..  ..  ..
390   M  20          2   9   9   9
391   M  17          1  14  16  16
392   M  21          1  10   8   7
393   M  18          1  11  12  10
394   M  19          1   8   9   9

[395 rows x 6 columns]

# 27- Dans la mini base, ajoutons 2 points sur la note de maths de tous les eleves

In [118]:
mini_base["Maths"] = mini_base["Maths"] +2

mini_base

sex age  studytime  Maths  Anglais  Français
0     F  18          2      7        6         6
1     F  17          2      7        5         6
2     F  18          2      9        8        10
3     F  15          3     17       14        15
4     F  16          2      8       10        10
..   ..  ..        ...    ...      ...       ...
390   M  20          2     11        9         9
391   M  17          1     16       16        16
392   M  21          1     12        8         7
393   M  18          1     13       12        10
394   M  19          1     10        9         9

[395 rows x 6 columns]

# 28-Dans la mini base ,remplacons dans la colonne sex, F par feminin et M par masculin

In [131]:
mini_base["sex"].replace({"F":"féminin","M":"masculin"})

0       féminin
1       féminin
2       féminin
3       féminin
4       féminin
         ...   
390    masculin
391    masculin
392    masculin
393    masculin
394    masculin
Name: sex, Length: 395, dtype: object

# 29- Ajoutons dans la mini base une colonne moyenne des notes de chaque etudiant

# 30-Dans le dataset d'origine ajouter une colonne "parent_job" qui contiendra le metiers des deux parents de chaque eleve concatene